flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/FlinkUtils.java

```diff
@@ -403,6 +403,28 @@ public static Long calculateClusterMemoryUsage(Configuration conf, int taskManag
         return tmTotalMemory + jmTotalMemory;
     }
 
+    public static Long calculateClusterStateSize(Configuration conf, int taskManagerReplicas) {
```

Comment:
Shouldn't this be called totalClusterMemorySize instead of state?

Reply:
Also I don't think this is used anywhere... :)

Reply:
State size or checkpoint size isn't directly related to the cluster memory size. For the heap memory backend, we would expect the state size to be lower than the overall memory. For RocksDB, it could even exceed the cluster memory.

Reply:
Ok but I still don't get 3 things:
 - Where is this used?
 - Why do we need this bad approximation if state size metrics are available from Flink?
 - This is basically just total memory, why do we call it state size?

Reply:
I didn't see your comment in https://github.com/apache/flink-kubernetes-operator/pull/941#discussion_r1943008300, it was somehow hidden when I replied.

Sorry, this code was unused code. I have removed it.